apps/src/code-studio/pd/application/teacher/ProfessionalLearningProgramRequirements.jsx

```diff
@@ -99,78 +190,10 @@ const ProfessionalLearningProgramRequirements = props => {
-    } else if (regionalPartner === null) {
-      return (
-        <>
-          <p>
-            <strong>
-              There is no Regional Partner in your region at this time
-            </strong>
-          </p>
-          <p>
-            Code.org will review your application and contact you with options
-            for joining the program hosted by a Regional Partner from a
-            different region. Please note that we are not able to guarantee a
-            space for you with another Regional Partner, and you will be
-            responsible for the costs associated with traveling to that location
-            if a virtual option is not available.
-          </p>
-        </>
-      );
+    } else if (hasNoRegionalPartner()) {
+      return renderProgramRequirements(false);
     } else {
-      // regional partner exists and is not errored
-      return (
-        <div>
-          <p>
-            Code.org’s Professional Learning Program is a yearlong program
-            starting in the summer and concluding in the spring. Workshops can
-            be held in-person, virtually, or as a combination of both throughout
-            the year. Refer to {`${regionalPartner.name}'s `}
-            <a
-              href={
-                pegasus('/educate/professional-learning/program-information') +
-                (!!data.schoolZipCode ? '?zip=' + data.schoolZipCode : '')
-              }
-              target="_blank"
-              rel="noopener noreferrer"
-            >
-              landing page
-            </a>{' '}
-            for more information about the schedule and delivery model.
-          </p>
-          <LabeledRadioButtonsWithAdditionalTextFields
-            name="committed"
-            textFieldMap={{
-              [TextFields.noExplain]: 'other'
-            }}
-          />
-          {renderAssignedWorkshopList()}
-          <div>
-            <label>
-              {regionalPartner.name} may have scholarships available to cover
-              the cost of the program.{' '}
-              <a
-                href={
-                  pegasus(
-                    '/educate/professional-learning/program-information'
-                  ) + (!!data.schoolZipCode ? '?zip=' + data.schoolZipCode : '')
-                }
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                Click here to check the fees and discounts for your program
-              </a>
-              . Let us know if your school or district would be able to pay the
-              fee or if you need to be considered for a scholarship.
-            </label>
-            <LabeledSingleCheckbox name="understandFee" />
-            <LabeledRadioButtons name="payFee" />
-            {data.payFee === TextFields.noPayFee && (
-              <LabeledLargeInput name="scholarshipReasons" />
-            )}
-          </div>
-        </div>
-      );
+      return renderProgramRequirements(true);
```

Comment:
maybe collapse these two cases into a single case, and inside `renderProgramRequirements` just reference the `regionalPartner` variable instead of using a parameter (the scope of that function has access to all the variables at the top)

Reply:
💯